discord/channel.py

```diff
@@ -1808,6 +1810,199 @@ async def edit(self, *, reason=None, **options):
         if payload is not None:
             # the payload will always be the proper channel payload
             return self.__class__(state=self._state, guild=self.guild, data=payload)  # type: ignore
+        
+
+class ForumChannel(discord.abc.GuildChannel, Hashable):
+    """Represents a Discord channel forum."""
+
+    __slots__ = (
+        'name',
+        'id',
+        'guild',
+        '_state',
+        'nsfw',
+        'category_id',
+        'position',
+        'slowmode_delay',
+        '_overwrites',
+        '_type',
+        'default_auto_archive_duration',
+    )
+
+    def __init__(self, *, state: ConnectionState, guild: Guild, data: ForumChannelPayload):
+        self._state: ConnectionState = state
+        self.id: int = int(data["id"])
+        self._type: int = data['type']
+        self._update(guild, data)
+
+    def __repr__(self) -> str:
+        return f"<ForumChannel id={self.id} name={self.name!r} position={self.position} nsfw={self.nsfw}>"
+
+    def _update(self, guild: Guild, data: ForumChannelPayload) -> None:
+        self.guild: Guild = guild
+        self.name: str = data["name"]
+        self.category_id: Optional[int] = utils._get_as_snowflake(data, "parent_id")
+        self.nsfw: bool = data.get("nsfw", False)
+        self.position: int = data["position"]
+        self.slowmode_delay: int = data.get('rate_limit_per_user', 0)
+        self.default_auto_archive_duration: ThreadArchiveDuration = data.get('default_auto_archive_duration', 1440)
+        self._type: int = data.get('type', self._type)
```

Comment:
Why does it set `_type` twice?

Reply:
If you're referring to the copy in `__init__` and the copy in `_update()`, I was basing the code off the `TextChannel()` class: 

https://github.com/Pycord-Development/pycord/blob/master/discord/channel.py#L182

Reply:
I see, thanks for responding